### PR TITLE
added support for a reference mapping

### PR DESCRIPTION
**Pull request by stmichael**

### CODE REVIEW

**Comment by senny** [comment]
@stmichael What do you think about a default for `:on => :id`?

Another thing I noticed is that the hash-keys are not really explanatory what they mean. Some Ideas:
- rename :on to :lookup?
- keep the structure of a normal mapping. `reference 'ArticleOID' => :article_id, :definition => 'Articles', :lookup => 'reference'` ?

**Comment by stmichael** [comment]
@senny What if we go for something like `reference 'Articles', 'ArticleOID' => :article_id, :look_up => 'ref_nr'. This way you would instantly see that this is a reference to 'Articles', the attribute mapping would be the same as normal mappings and the rest would be optional parameters.

**Comment by senny** [comment]
@stmichael sounds good. Can you name the key `:lookup` instead of `:look_up`? Just to be consistent across the api.

**Comment by stmichael** [comment]
@senny nope ;-)

**Comment by stmichael** [comment]
@senny what do you think about that?

File: lib/data-import/definition/mappings.rb [hunk 1]
```diff
@@ -30,6 +30,22 @@ def apply!(definition, context, row, output_row)
...
+    class ReferenceMapping
+      def initialize(referenced_definition, options = {})
+        raise ArgumentError, 'you must at least specify the mapping of the foreign key' if options.empty?
+
+        @referenced_definition = referenced_definition
+        @lookup_name = options.delete(:lookup) || :id
+        foreign_key = options.keys.first
+        @old_foreign_key = foreign_key.to_sym
+        @new_foreign_key = options[foreign_key].to_sym
+      end
```
Comment (senny) [inline lib/data-import/definition/mappings.rb]: I would revert this code to the previous version, and have the dsl decide what options to pass along. Also I would put the default in the construction part of the dsl and not in the definition itself.

File: spec/acceptance/references_spec.rb [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+      mapping 'OID' => :id
+    end
+
+    import 'person' do
+      dependencies 'address'
+
+      from 'Person'
+      to 'people'
+
+      reference 'address', 'AddressOID' => :address_id
```
Comment (senny) [inline spec/acceptance/references_spec.rb:23]: I think you should add a test-case for the lookup option.

**Comment by senny** [comment]
I added a few minor comments. The rest looks good!

**Comment by stmichael** [comment]
@senny how about now?

**Comment by senny** [comment]
nice, merged!

**Pull request merged**